ui-spacetimechart: use paths layer for conflicts

emersion opened the pull request:
The overlay layer is rendered on top of the captions, which is undesirable here.

Example bug:

![out](https://github.com/user-attachments/assets/2726cb6d-d020-4538-9927-7c4ca5795b1e)

Akctarus reviewed (approved):
LGTM

SharglutDev reviewed (commented):
Doesn't seem like it fixes the issue ? Or did I do something wrong ?

![Capture d’écran 2024-10-17 à 09 57 56](https://github.com/user-attachments/assets/2faa2b87-a992-4e73-aaa7-09fc918c956e)

clarani commented:
I still have the issue too :/

emersion committed:
ui-spacetimechart: use paths layer for conflicts

The overlay layer is rendered on top of the captions, which is
undesirable here.

Signed-off-by: Simon Ser <contact@emersion.fr>

emersion commented:
Apologies, I messed up when testing. I've updated the PR. Here's a hack patch to test this in the storybook:

```diff
diff --git a/ui-spacetimechart/src/stories/layers.stories.tsx b/ui-spacetimechart/src/stories/layers.stories.tsx
index 028ee5b5f118..2c6479c0f5d3 100644
--- a/ui-spacetimechart/src/stories/layers.stories.tsx
+++ b/ui-spacetimechart/src/stories/layers.stories.tsx
@@ -32,7 +32,7 @@ const CONFLICTS = [
     timeStart: +START_DATE + 35 * MINUTE,
     timeEnd: +START_DATE + 37 * MINUTE,
     spaceStart: 39 * KILOMETER,
-    spaceEnd: 41 * KILOMETER,
+    spaceEnd: 99999 * KILOMETER,
   },
 ];
 
```

SharglutDev reviewed (approved):
Lgtm and tested. Should we add another commit to do the same for occupation blocks ?

emersion commented:
That one uses the `background` layer, because it's drawn behind the paths.
